lib/plugins/rotate_asg_instances.rb

```diff
@@ -7,11 +7,41 @@ module Moonshot
   module Plugins
     # Rotate ASG instances after update.
     class RotateAsgInstances
-      def post_update(resources)
-        asg = ASG.new(resources)
+      include ::Moonshot::DoctorHelper
+
+      def initialize(ssh_user = ENV['MOONSHOT_SSH_USER'])
+        @ssh_user = ssh_user
+      end
+
+      def doctor(resources)
+        @resources = resources
+        doctor_check_ssh
+      end
+
+      def pre_update(resources)
+        @resources = resources
+        asg.try_to_ssh(raise_exception: true)
+      end
+
+      def post_update(_resources)
         asg.rotate_asg_instances
         asg.teardown_outdated_instances
       end
+
+      private
+
+      def doctor_check_ssh
+        if asg.try_to_ssh
+          success('Successfully opened SSH connection to a worker instance.')
```

Comment:
I think "worker instance" is related to Cloud Data service, we should say only "instance"